### PR TITLE
[Do not merge but review] Draft for Migration guide to CS 2.0

### PR DESCRIPTION
**Pull request by kuenishi**

### CODE REVIEW

File: source/languages/en/riakcs/upgrade-v20.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-x.y.z/ebin"]}
+]}.
+```
+
+If multibag is to be used, add multibag ebin path:
+
+```erlang
+{riak_kv, [
+  {add_paths, ["/usr/lib/riak-cs/lib/riak_cs-x.y.z/ebin",
+               "/usr/lib/riak-cs/lib/riak_cs_multibag-a.b.c/ebin"]}
```
Comment (shino) [inline source/languages/en/riakcs/upgrade-v20.md]: multibag's ebin is not needed.

File: source/languages/en/riakcs/upgrade-v20.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+Consult
+[Configuration Mapping Table](https://github.com/basho/riak_cs/wiki/Configuration-Mapping-Table-between-1.5-and-2.0-%5BRFC%5D)
+to preserve same configuration between CS 1.5 and 2.0.
+
+If obsolete configurations like `fold_objects_for_list_keys`,
+`n_val_1_get_requests` or `gc_paginated_indexes` are still set as
+`false`, **removing any of them from configuraiton file** is strongly
+recommended at Riak CS 2.0, because those configuration items are to
+preserve old and slow behaviours of Riak CS and have no bad effect on
```
Comment (shino) [inline source/languages/en/riakcs/upgrade-v20.md]: s/no bad/bad/ ?

**Comment by shino** [comment]
Connectivity matrix helps?

```

  | connecter \ connectee | Riak 1.4  | Riak 2.0  | Stanchion 1.5 | Stanchion 2.0 |
  |-----------------------+-----------+-----------+---------------+---------------|
  | Stanchion 1.5         | OK        | probably? | N/A           | N/A           |
  | Stanchion 2.0         | probably? | OK        | N/A           | N/A           |
  | Riak CS 1.5           | OK        | probably? | OK            | OK?           |
  | Riak CS 2.0           | probably? | OK        | OK?           | OK            |
```

File: source/languages/en/riakcs/upgrade-v20.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+2. Stop Riak process
+3. Backup all configuration files
+4. Upgrade Riak CS software
+5. Upgrade Riak software
+6. Update Riak configuration
+7. Migrate Riak CS configuration
+8. Start Riak
+9. Start Riak CS
+
+Stanchion can be updated at any time during the system upgrade. Be
```
Comment (shino) [inline source/languages/en/riakcs/upgrade-v20.md]: Will we test both stanchion1.5-riak2.0 and stanchion2.0-riak1.4?

File: source/languages/en/riakcs/upgrade-v20.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+cannot create any bucket after upgrade unless the limit number
+extended at the system configuration.
+
+## Special Notes on Upgrading From Riak CS 1.4
+
+An operational procedure
+[to clean up incomplete multipart under deleted buckets](https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#notes-on-upgrading-2)
+is needed. Otherwise new buckets which used to exist in the past
+couldn't be created. The operation will fail with 409 Conflict.
+
```
Comment (shino) [inline source/languages/en/riakcs/upgrade-v20.md]: Is it worthwhile to mention Difference of GC timestamp management?

File: source/languages/en/riakcs/upgrade-v20.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+## Riak CS nodes not corriding in a same box with live Riak node
+
+Riak CS nodes not corriding in a same box with live Riak node can be
+upgraded at any time while corresponding remote Riak is alive.
```
Comment (shino) [inline source/languages/en/riakcs/upgrade-v20.md]: Are both CS2.0-to-Riak1.4 and CS1.5-to-Riak2.0?

I'm thinking about the configurtion where Riak CS connects to
(potentially many) Riak via L4 reverse proxy like HAProxy.

**Comment by kuenishi** [comment]
This is superseded by https://github.com/basho/riak_cs/pull/1082. Discussion will continue there.

**Pull request closed without merging**